src/main/java/com/commercetools/sync/producttypes/ProductTypeSync.java

```diff
@@ -160,7 +160,7 @@ private boolean validateDraft(@Nullable final ProductTypeDraft draft) {
      * @param <T>          a type that extends of {@link WithKey}.
      * @return the map of keys to {@link ProductType}/{@link ProductTypeDraft} instances.
      */
-    private <T extends WithKey> Map<String, T> getKeysProductTypeMap(@Nonnull final List<T> productTypes) {
+    private <T extends WithKey> Map<String, T> getProductTypeKeysMap(@Nonnull final Set<T> productTypes) {
```

Comment:
Good catch! https://github.com/commercetools/commercetools-sync-java/pull/316/commits/6730353672aa18681ccfe3bf6ca019f74d8f0526

Reply:
Right now, if the set contains two productTypes with the same key, this method will choose one of the duplicates and disregard the other. I think we should just use the default merging function:
````java
.collect(Collectors.toMap(ProductType::getKey, Function.identity()));
````
 which throws an exception in that case, or maybe we should document this behaviour.

Reply:
Definitely, this is better than just skipping it. It seems this is related [this](https://github.com/commercetools/commercetools-sync-java/pull/316#discussion_r229707233) with maybe this method is not needed.

Reply:
This code snippet creates a map old product types which this list coming from ctp(**fetchMatchingProductTypesByKeys**), it seems unnecessary to check merge function because we will not get throwing a duplicate key exception. So I moved this check to the line where this function called and remove method in this commit https://github.com/commercetools/commercetools-sync-java/pull/316/commits/1f2d40e9908e85661a475ef1d8b1bcc7430f968a. Wdyt?